R/in-notin.R

```diff
@@ -22,14 +27,21 @@
 '%in%' <- function(x, y) {
   if (is.character(x) && is.character(y)) {
     x %chin% y
-  } else if (is.list(y)) {
-    # https://github.com/markfairbanks/tidytable/issues/565
-    base::'%in%'(x, y)
-  } else {
+  } else if (vec_ptype_compatible(x, y)) {
     vec_in(x, y)
+  } else {
+    base::'%in%'(x, y)
   }
 }
 
+#' check if inputs are not compatible with vec_in for the fast path of %in%
+#' @noRd
+needs_base_in <- function(x, y) {
```

Comment:
Oh whoops I didn't call out that this needed to be changed. The `vec_ptype_compatible()` functions used above is an internal helper. You'll want to paste the whole function in as it flips the output compared to `needs_base_in()`

```r
# Check if two vectors have compatible ptypes
vec_ptype_compatible <- function(x, y) {
  # https://github.com/markfairbanks/tidytable/issues/565
  # https://github.com/markfairbanks/tidytable/issues/632
  tryCatch({vec_ptype_common(x, y); TRUE}, error = function(e) FALSE)
}
```